TACC/Core-CMS-Resources
# feat: tup-87 social media share links

## Overview

Support social media share links on Texascale via snippet.

## Related

- [TUP-87](https://jira.tacc.utexas.edu/browse/TUP-87)
- requires https://github.com/TACC/Core-CMS/pull/725

## Changes

- **added** share links on article templates

## Testing

> **Note**
> Deployed to staging. See any article, such as [`/2022/…/forecasting-future-climate/`](https://pprd.texascale.tacc.utexas.edu/2022/feature-stories/forecasting-future-climate/).

1. Create page.
2. Set Template to "Article".
3. Verify social media share links appear.
4. Verify social media share links function.
5. Set Template to "Article (Free-Form)".
6. Add Columns and Text to mirror [staging `/2022/…/forecasting-future-climate/`](https://pprd.texascale.tacc.utexas.edu/2022/feature-stories/forecasting-future-climate/).
7. Verify social media share links appear.
8. Verify social media share links function.

## UI

| no hover | hover Facebook | hover LinkedIn | hover E-mail |
| - | - | - | - |
| ![local - no hover](https://github.com/TACC/Core-CMS-Resources/assets/62723358/cbf449e2-b1d0-4602-b606-5519b0b15d52) | ![local - hover fb](https://github.com/TACC/Core-CMS-Resources/assets/62723358/f8c77a5e-348e-4119-9f1c-6a02d1503dd7) | ![local - hover ln](https://github.com/TACC/Core-CMS-Resources/assets/62723358/a0ac0023-29e1-4c54-950b-a0c76799d3fb) | ![local - hover em](https://github.com/TACC/Core-CMS-Resources/assets/62723358/0315f14b-71cc-4ea4-a55e-59fdac6f2a7f) |

<details><summary>Archived</summary>

| no hover | hover Facebook | hover LinkedIn |
| - | - | - |
| ![no hover](https://github.com/TACC/Core-CMS-Resources/assets/62723358/d63f9dab-b966-47fb-af17-362851b2af54) | ![hover fb](https://github.com/TACC/Core-CMS-Resources/assets/62723358/87634778-c919-424b-af8c-6bdf167a6212) | ![hover ln](https://github.com/TACC/Core-CMS-Resources/assets/62723358/203698d6-ef85-46f2-a32b-1ac6acc5df44) |

| Unconfirmed Design |
| - |
| <img width="992" alt="tup-87 texscale" src="https://github.com/TACC/Core-CMS-Resources/assets/62723358/2d02ce90-5e1d-470b-8bb7-44d22512656b"> |

https://github.com/TACC/Core-CMS-Resources/assets/62723358/e5b1718c-116a-4228-98db-5070b9656f6f

| Proof of Concept |
| - |
| <img width="1200" alt="poc texascale socail media" src="https://github.com/TACC/Core-CMS-Resources/assets/62723358/350ba72f-d86b-4228-85a5-e37572878814"> |

</details>
